### ISSUE TITLE
`maxlength` < 16 breaks .clear()

### ISSUE THREAD

**Issue by HugoGiraudel** [cypress-io/cypress#955]
- Operating System: 10.12.6
- Cypress Version: 1.1.1
- Browser Version: n/a

### Is this a Feature or Bug?

A bug.

### Current behavior:

Calling `.clear()` on an input with a `maxlength` attribute smaller than 16 outputs odd characters in the field instead of clearing it.

### Desired behavior:

Calling `.clear()` on an input with a `maxlength` attribute smaller than 16 clears the field.

#### Test code:

```js
cy.get('input[maxlength]').clear()
```

### Additional Info (images, stack traces, etc)

This bug was introduced in [1.1.1](https://docs.cypress.io/guides/references/changelog.html#1-1-1) with https://github.com/cypress-io/cypress/pull/930 as mentioned in [this comment](https://github.com/cypress-io/cypress/issues/928#issuecomment-346369782).

Calling `.clear()` seems to call `.type()` with `{selectall}{del}` as per the following snippet. This string is 16 characters long and takes into account the `maxlength` attribute (while it should not).

https://github.com/cypress-io/cypress/blob/cb67a5ace3a9ff8d6bb7b5d93f06048dd3469ec5/packages/driver/src/cy/commands/actions/type.coffee#L402

The changes introduced in 1.1.1 slice the characters to respect the `maxlength` attribute value, therefore slicing these tokens. This is why a maxlength from 1 to 15 causes this bug.

I guess not counting tokens in the characters count is probably the way to go. Pseudo code could look like:

```js
const tokens = RegExp(Object.keys(keyStandardMap).join('|'), 'g')
chars.replace(tokens, '')
```

**Comment by brian-mann** [comment]
The fix for this is unfortunately much more complicated. We'll need to also continue to fire all applicable events minus the `input` event past `maxlength` but not change the value of the input.

Didn't want to work on this now but since this is a regression we'll get out a fix.

**Comment by brian-mann** [comment]
Doing the work here https://github.com/cypress-io/cypress/pull/961

**Comment by brian-mann** [comment]
Released in [`1.1.2`](https://docs.cypress.io/guides/references/changelog.html#1-1-2).
